### PR TITLE
[WIP] feat(raiko): enable sqlite db by feature

### PR DESCRIPTION
**Pull request by smtmfft**
Try to solve the lifetime issue, solution maybe far away from perfect as the db connection lifetime is hard to handle....

### CODE REVIEW

File: task_manager/Cargo.toml [hunk 1]
```diff
@@ -22,9 +22,13 @@ rand = "0.9.0-alpha.1"                                            # This is an a
 rand_chacha = "0.9.0-alpha.1"
 tempfile = "3.10.1"
 alloy-primitives = { workspace = true, features = ["getrandom"] }
-
 rusqlite = { workspace = true, features = ["trace"] }
```
Comment (petarvujovic98) [inline task_manager/Cargo.toml:25]: This should become an optional feature enabled only when the `sqlite` feature is active

File: task_manager/src/adv_sqlite.rs
Comment (petarvujovic98) [inline task_manager/src/adv_sqlite.rs:1]: We should also return the prover row from queries in order to not just return a `None`. I'll change the queries and function wrappers to do this

**Review by petarvujovic98 — CHANGES_REQUESTED** [review]

**Comment by petarvujovic98** [comment]
@smtmfft Unrelated to the code, why is this branch on your fork instead of a new branch inside `raiko`?

File: task_manager/Cargo.toml (hunk 1 above)
Comment (smtmfft) [reply task_manager/Cargo.toml:25]: Not sure if later we need to use run-time flag or just using compile-time feature to choose the db impl. but yes, should be linked to optional sqlite now.

**Comment by smtmfft** [comment]
> @smtmfft Unrelated to the code, why is this branch on your fork instead of a new branch inside `raiko`?

oh, I get used to use mine when the target is not the main, will change for sure.

**Pull request closed without merging**